src/backend/libc/net/addr.rs

```diff
@@ -123,17 +143,23 @@ impl SocketAddrUnix {
     #[cfg(linux_kernel)]
     #[inline]
     pub fn abstract_name(&self) -> Option<&[u8]> {
-        let len = self.len();
-        if len != 0 && self.unix.sun_path[0] == 0 {
-            let end = len as usize - offsetof_sun_path();
+        let end = self.len().saturating_sub(offsetof_sun_path());
```

Comment:
Can you comment on why you changed this to use `saturating_sub`?

Reply:
I wanted to remove one indentation level by removing the test if `len == 0`, followed by `len > offsetof_sun_path`. I don't know if there actually is a case where `len == 0`. If there isn't, then a normal subtraction would be good enough. Alternatively, because the method returns an `Option`, `checked_sub` would work, too.